src/Compilers/Test/Utilities/CSharp/CSharpTestBase.cs

```diff
@@ -472,23 +437,48 @@ public static SyntaxTree ParseWithRoundTripCheck(string text, CSharpParseOptions
             IEnumerable<MetadataReference> references = null,
             CSharpCompilationOptions options = null,
             CSharpParseOptions parseOptions = null,
-            string assemblyName = "")
-        {
-            return CreateCompilation(new[] { Parse(source, options: parseOptions) }, references, options, assemblyName);
-        }
+            TargetFramework targetFramework = TargetFramework.Standard,
+            string assemblyName = "",
+            string sourceFileName = "") => CreateCompilationWithNone(new[] { Parse(source, sourceFileName, parseOptions) }, TargetFrameworkUtil.GetReferences(targetFramework, references), options, assemblyName);
 
         public static CSharpCompilation CreateCompilation(
-            IEnumerable<string> sources,
+            IEnumerable<string> source,
             IEnumerable<MetadataReference> references = null,
             CSharpCompilationOptions options = null,
             CSharpParseOptions parseOptions = null,
-            string assemblyName = "")
-        {
-            return CreateCompilation(Parse(sources, parseOptions), references, options, assemblyName);
-        }
+            TargetFramework targetFramework = TargetFramework.Standard,
+            string assemblyName = "") => CreateCompilationWithNone(Parse(source, parseOptions), TargetFrameworkUtil.GetReferences(targetFramework, references), options, assemblyName);
+
+        public static CSharpCompilation CreateCompilation(
+            SyntaxTree source,
+            IEnumerable<MetadataReference> references = null,
+            CSharpCompilationOptions options = null,
+            TargetFramework targetFramework = TargetFramework.Standard,
+            string assemblyName = "") => CreateCompilationWithNone(new[] { source }, TargetFrameworkUtil.GetReferences(targetFramework, references), options, assemblyName);
 
         public static CSharpCompilation CreateCompilation(
-            IEnumerable<SyntaxTree> trees,
+            IEnumerable<SyntaxTree> source,
+            IEnumerable<MetadataReference> references = null,
+            CSharpCompilationOptions options = null,
+            TargetFramework targetFramework = TargetFramework.Standard,
+            string assemblyName = "") => CreateCompilationWithNone(source, TargetFrameworkUtil.GetReferences(targetFramework, references), options, assemblyName);
+
+        public static CSharpCompilation CreateCompilationWithNone(
```

Comment:
Can we call this `CreateEmptyCompilation`? I think it makes a lot more sense out of context.

Reply:
I struggled a bunch with what to call this particular overload. The `WithNone` suffix makes it consistent with all of the other overloads in having a `With<TargetFramework value>` pattern. But I'm not sure it's particularly readable.